crates/query-engine/metadata/src/metadata/native_queries.rs

```diff
@@ -55,62 +197,38 @@ pub enum NativeQueryPart {
     Parameter(String),
 }
 
-/// A Native Query SQL after parsing.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct NativeQuerySql(pub Vec<NativeQueryPart>);
+/// A Native Query SQL parts after parsing.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(from = "String")]
+#[serde(into = "String")]
+pub struct NativeQueryParts(pub Vec<NativeQueryPart>);
 
-// Serialization
+impl From<String> for NativeQueryParts {
+    /// Used for de-serialization.
+    fn from(value: String) -> Self {
+        parse_native_query(&value)
+    }
+}
 
-impl Serialize for NativeQuerySql {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
+impl From<NativeQueryParts> for String {
+    /// Used for serialization.
+    fn from(value: NativeQueryParts) -> Self {
         let mut sql: String = String::new();
-        for part in self.0.iter() {
+        for part in value.0.iter() {
             match part {
                 NativeQueryPart::Text(text) => sql.push_str(text.as_str()),
                 NativeQueryPart::Parameter(param) => {
                     sql.push_str(format!("{{{{{}}}}}", param).as_str())
                 }
             }
         }
-        serializer.serialize_str(&sql)
-    }
-}
-
-struct NQVisitor;
-
-impl<'de> serde::de::Visitor<'de> for NQVisitor {
-    type Value = NativeQuerySql;
-
-    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
-        formatter.write_str("A Native Query SQL")
-    }
-
-    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
-    where
-        E: serde::de::Error,
-    {
-        // todo: write a better parser
-        let sql = parse_native_query(value);
-
-        Ok(NativeQuerySql(sql))
+        sql
     }
 }
 
-impl<'de> Deserialize<'de> for NativeQuerySql {
-    fn deserialize<D>(deserializer: D) -> Result<NativeQuerySql, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_str(NQVisitor)
-    }
-}
-
-impl JsonSchema for NativeQuerySql {
+impl JsonSchema for NativeQueryParts {
```

Comment:
You can probably also avoid this instance with a `#[schemars(rename=..)]` attribute.

Reply:
I think this doesn't work because it attempts to create a schema for `Vec<NativeQueryPart>`, even if I add `with = "String"`.